test/integration/node-specific/client_close.test.ts

```diff
@@ -0,0 +1,300 @@
+
+import { type TestConfiguration } from '../../tools/runner/config';
+import { runScriptAndGetProcessInfo } from './resource_tracking_script_builder';
+
+describe.skip('MongoClient.close() Integration', () => {
+  // note: these tests are set-up in accordance of the resource ownership tree
+
+  let config: TestConfiguration;
+
+  beforeEach(function () {
+    config = this.configuration;
+  });
+
+  describe('Node.js resource: TLS File read', () => {
+    describe('when client is connecting and reads an infinite TLS file', () => {
+      it('the file read is interrupted by client.close()', async function () {
+        await runScriptAndGetProcessInfo(
+          'tls-file-read',
+          config,
+          async function run({ MongoClient, uri, expect }) {
+            const infiniteFile = '/dev/zero';
+            const client = new MongoClient(uri, { tlsCertificateKeyFile: infiniteFile });
+            client.connect();
+            expect(process.getActiveResourcesInfo()).to.include('FSReqPromise');
+            await client.close();
+            expect(process.getActiveResourcesInfo()).to.not.include('FSReqPromise');
+          }
+        );
+      });
+    });
+  });
+
+  describe('MongoClientAuthProviders', () => {
+    describe('Node.js resource: Token file read', () => {
+      let tokenFileEnvCache;
+
+      beforeEach(function () {
+        if (process.env.AUTH === 'auth') {
+          this.currentTest.skipReason = 'OIDC test environment requires auth disabled';
+          return this.skip();
+        }
+        tokenFileEnvCache = process.env.OIDC_TOKEN_FILE;
+      });
+
+      afterEach(function () {
+        process.env.OIDC_TOKEN_FILE = tokenFileEnvCache;
+      });
+
+      describe('when MongoClientAuthProviders is instantiated and token file read hangs', () => {
+        it('the file read is interrupted by client.close()', async () => {
+          await runScriptAndGetProcessInfo(
+            'token-file-read',
+            config,
+            async function run({ MongoClient, uri, expect }) {
+              const infiniteFile = '/dev/zero';
+              process.env.OIDC_TOKEN_FILE = infiniteFile;
+              const options = {
+                authMechanismProperties: { ENVIRONMENT: 'test' },
+                authMechanism: 'MONGODB-OIDC'
+              };
+              const client = new MongoClient(uri, options);
+              client.connect();
+              expect(process.getActiveResourcesInfo()).to.include('FSReqPromise');
+              await client.close();
+              expect(process.getActiveResourcesInfo()).to.not.include('FSReqPromise');
+            }
+          );
+        });
+      });
+    });
+  });
+
+  describe('Topology', () => {
+    describe('Node.js resource: Server Selection Timer', () => {
+      describe('after a Topology is created through client.connect()', () => {
+        it('server selection timers are cleaned up by client.close()', async () => {});
```

Comment:
Can you search replace all the `it` with `it.skip` for the ones that are empty? I ask specifically so when we're taking the .skip off the describe and I'm running these on my end its easy to see which one's are passing because they are actually running vs passing because they are still empty.